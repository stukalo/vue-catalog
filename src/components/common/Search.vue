<template>
  <div class='search'>
    <input
      class='search_input'
      type="text"
      ref="search-input"
      autocomplete="off"
      spellcheck="false"
      placeholder="Search"
      @input="this.onInput"
      :value="search.value"
    />
    <Button
      class='search_button'
      @click="this.onSubmit"
    >
      Search
    </Button>
  </div>
</template>

<script>
import Button from './Button.vue';

export default {
  name: 'Search',
  components: {
    Button,
  },
  props: ['search'],
  methods: {
    onSubmit() {
      this.$emit('submit');
    },
    onInput(evt) {
      this.$emit('valueChange', evt.target.value);
    },
  },
};
</script>

<style scoped lang="less">
@import (reference) '../../assets/css/variables.less';
@import (reference) '../../assets/css/reset.less';

.search {
  min-width: 600px;
  display: flex;
  .fontMixin(25px);
}

.search_input {
  flex: 1;
  border: none;
  box-sizing: border-box;
  height: 60px;
  padding: 0 15px;
  margin: 0 14px 0 0;
  color: @primaryText;
  border-radius: @primaryBorderRadius;
  background-color: @primaryBackgroundOpacity;
  .fontMixin(22px);

  &:focus-visible {
    outline: none;
  }
}

</style>
