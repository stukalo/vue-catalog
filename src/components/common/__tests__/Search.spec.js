import { shallowMount } from '@vue/test-utils';
import Search from '../Search.vue';

const props = {
  search: {
    value: 'mock_value',
  },
  placeholder: 'mock_placeholder',
  buttonText: 'mock_button_text',
};

describe('Search', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Search);
    expect(wrapper)
      .toMatchSnapshot();
  });

  describe('Input attributes', () => {
    let input;
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(Search, {
        propsData: props,
      });
      input = wrapper.find('input');
    });

    it('should have correct placeholder', () => {
      expect(input.attributes().placeholder).toEqual(props.placeholder);
    });

    it('should have correct value', () => {
      expect(input.element.value).toEqual(props.search.value);
    });
  });
});
