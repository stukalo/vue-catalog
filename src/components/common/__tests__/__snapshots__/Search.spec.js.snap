// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="search"
      >
        <input
          autocomplete="off"
          class="search_input"
          placeholder="Search"
          spellcheck="false"
          type="text"
        />
        <button-stub
          class="search_button"
          type="button"
        />
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "buttonText": "Search",
          "hasOwnProperty": [Function],
          "onInput": [Function],
          "onSubmit": [Function],
          "placeholder": "Search",
          "search": Object {
            "value": "",
          },
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "buttonText": 3,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "onInput": 4,
            "onSubmit": 4,
            "placeholder": 3,
            "search": 3,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {
            "Button": Object {
              "name": "Button",
              "render": [Function],
            },
          },
          "ctx": Object {
            "buttonText": "Search",
            "hasOwnProperty": [Function],
            "onInput": [Function],
            "onSubmit": [Function],
            "placeholder": "Search",
            "search": Object {
              "value": "",
            },
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "buttonText": "Search",
            "placeholder": "Search",
            "search": Object {
              "value": "",
            },
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "buttonText": Object {
                "0": false,
                "1": true,
                "default": "Search",
              },
              "placeholder": Object {
                "0": false,
                "1": true,
                "default": "Search",
              },
              "search": Object {
                "0": false,
                "1": true,
                "default": Object {
                  "value": "",
                },
              },
            },
            Array [
              "search",
              "placeholder",
              "buttonText",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "buttonText": "Search",
            "hasOwnProperty": [Function],
            "onInput": [Function],
            "onSubmit": [Function],
            "placeholder": "Search",
            "search": Object {
              "value": "",
            },
          },
          "refs": Object {
            "search-input": <input
              autocomplete="off"
              class="search_input"
              placeholder="Search"
              spellcheck="false"
              type="text"
            />,
          },
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "placeholder",
                  "value",
                ],
                "el": <input
                  autocomplete="off"
                  class="search_input"
                  placeholder="Search"
                  spellcheck="false"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": Object {
                  "autocomplete": "off",
                  "class": "search_input",
                  "onInput": [Function],
                  "placeholder": "Search",
                  "ref": "search-input",
                  "spellcheck": "false",
                  "type": "text",
                  "value": "",
                },
                "ref": Object {
                  "i": [Circular],
                  "r": "search-input",
                },
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Object {
                  "default": [Function],
                },
                "component": Object {
                  "a": null,
                  "accessCache": Object {},
                  "appContext": Object {
                    "app": [Circular],
                    "components": Object {
                      "transition": Object {
                        "name": "transition",
                      },
                      "transition-group": Object {
                        "name": "transition-group",
                      },
                    },
                    "config": Object {
                      "compilerOptions": Object {},
                      "errorHandler": undefined,
                      "globalProperties": Object {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": Object {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": Object {},
                    "emitsCache": WeakMap {},
                    "mixins": Array [
                      Object {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": Object {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": Object {
                    "class": "search_button",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": Object {},
                  "da": null,
                  "data": Object {},
                  "directives": null,
                  "ec": null,
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": Object {},
                  "propsDefaults": Object {},
                  "propsOptions": Array [
                    Object {},
                    Array [],
                  ],
                  "provides": Object {},
                  "proxy": Object {},
                  "refs": Object {},
                  "render": [Function],
                  "renderCache": Array [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "active": true,
                    "cleanups": Array [],
                    "effects": Array [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "scheduler": [Function],
                      },
                    ],
                  },
                  "setupContext": null,
                  "setupState": Object {},
                  "slots": Object {
                    "default": [Function],
                  },
                  "sp": null,
                  "subTree": Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button-stub
                      class="search_button"
                      type="button"
                    />,
                    "key": null,
                    "patchFlag": 16,
                    "props": Object {
                      "class": "search_button",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": Object {
                    "compatConfig": Object {
                      "MODE": 3,
                      "RENDER_FUNCTION": false,
                    },
                    "name": "Button",
                    "props": Object {},
                    "render": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "onClick",
                ],
                "el": <button-stub
                  class="search_button"
                  type="button"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "class": "search_button",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 36,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Object {
                  "compatConfig": Object {
                    "MODE": 3,
                    "RENDER_FUNCTION": false,
                  },
                  "name": "Button",
                  "props": Object {},
                  "render": [Function],
                },
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "placeholder",
                  "value",
                ],
                "el": <input
                  autocomplete="off"
                  class="search_input"
                  placeholder="Search"
                  spellcheck="false"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": Object {
                  "autocomplete": "off",
                  "class": "search_input",
                  "onInput": [Function],
                  "placeholder": "Search",
                  "ref": "search-input",
                  "spellcheck": "false",
                  "type": "text",
                  "value": "",
                },
                "ref": Object {
                  "i": [Circular],
                  "r": "search-input",
                },
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Object {
                  "default": [Function],
                },
                "component": Object {
                  "a": null,
                  "accessCache": Object {
                    "@@__IMMUTABLE_ITERABLE__@@": 0,
                    "@@__IMMUTABLE_RECORD__@@": 0,
                    "_isMockFunction": 0,
                    "constructor": 0,
                    "isVueInstance": 0,
                    "toJSON": 0,
                    Symbol(Symbol.toStringTag): 0,
                  },
                  "appContext": Object {
                    "app": [Circular],
                    "components": Object {
                      "transition": Object {
                        "name": "transition",
                      },
                      "transition-group": Object {
                        "name": "transition-group",
                      },
                    },
                    "config": Object {
                      "compilerOptions": Object {},
                      "errorHandler": undefined,
                      "globalProperties": Object {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": Object {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": Object {},
                    "emitsCache": WeakMap {},
                    "mixins": Array [
                      Object {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": Object {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": Object {
                    "class": "search_button",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": Object {},
                  "da": null,
                  "data": Object {},
                  "directives": null,
                  "ec": null,
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": Object {},
                  "propsDefaults": Object {},
                  "propsOptions": Array [
                    Object {},
                    Array [],
                  ],
                  "provides": Object {},
                  "proxy": Object {},
                  "refs": Object {},
                  "render": [Function],
                  "renderCache": Array [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "active": true,
                    "cleanups": Array [],
                    "effects": Array [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "scheduler": [Function],
                      },
                    ],
                  },
                  "setupContext": null,
                  "setupState": Object {},
                  "slots": Object {
                    "default": [Function],
                  },
                  "sp": null,
                  "subTree": Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button-stub
                      class="search_button"
                      type="button"
                    />,
                    "key": null,
                    "patchFlag": 16,
                    "props": Object {
                      "class": "search_button",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": Object {
                    "compatConfig": Object {
                      "MODE": 3,
                      "RENDER_FUNCTION": false,
                    },
                    "name": "Button",
                    "props": Object {},
                    "render": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "onClick",
                ],
                "el": <button-stub
                  class="search_button"
                  type="button"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "class": "search_button",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 36,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Object {
                  "compatConfig": Object {
                    "MODE": 3,
                    "RENDER_FUNCTION": false,
                  },
                  "name": "Button",
                  "props": Object {},
                  "render": [Function],
                },
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="search"
            >
              <input
                autocomplete="off"
                class="search_input"
                placeholder="Search"
                spellcheck="false"
                type="text"
              />
              <button-stub
                class="search_button"
                type="button"
              />
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "search",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {
              "Button": Object {
                "name": "Button",
                "render": [Function],
              },
            },
            "methods": Object {
              "onInput": [Function],
              "onSubmit": [Function],
            },
            "name": "Search",
            "props": Object {
              "buttonText": Object {
                "0": false,
                "1": true,
                "default": "Search",
              },
              "placeholder": Object {
                "0": false,
                "1": true,
                "default": "Search",
              },
              "search": Object {
                "0": false,
                "1": true,
                "default": Object {
                  "value": "",
                },
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="search"
        >
          <input
            autocomplete="off"
            class="search_input"
            placeholder="Search"
            spellcheck="false"
            type="text"
          />
          <button-stub
            class="search_button"
            type="button"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "i": [Circular],
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {
            "Button": Object {
              "name": "Button",
              "render": [Function],
            },
          },
          "methods": Object {
            "onInput": [Function],
            "onSubmit": [Function],
          },
          "name": "Search",
          "props": Object {
            "buttonText": Object {
              "0": false,
              "1": true,
              "default": "Search",
            },
            "placeholder": Object {
              "0": false,
              "1": true,
              "default": "Search",
            },
            "search": Object {
              "0": false,
              "1": true,
              "default": Object {
                "value": "",
              },
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="search"
        >
          <input
            autocomplete="off"
            class="search_input"
            placeholder="Search"
            spellcheck="false"
            type="text"
          />
          <button-stub
            class="search_button"
            type="button"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.23",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "buttonText": "Search",
    "hasOwnProperty": [Function],
    "onInput": [Function],
    "onSubmit": [Function],
    "placeholder": "Search",
    "search": Object {
      "value": "",
    },
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <div
    class="search"
  >
    <input
      autocomplete="off"
      class="search_input"
      placeholder="Search"
      spellcheck="false"
      type="text"
    />
    <button-stub
      class="search_button"
      type="button"
    />
  </div>,
}
`;
